blockly-rtc/public/WorkspaceClient.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Class for managing client-server communication and event
+ * resolution.
+ * @author navil@google.com (Navil Perez)
+ */
+
+import {getEvents, writeEvents} from './api';
+
+/**
+ * Class for managing events between the workspace and the server.
+ * @param {string} workspaceId The id of the Blockly.Workspace instance this
+ * client corresponds to.
+ * */
+export default class WorkspaceClient {
+    constructor(workspaceId) {
+        this.workspaceId = workspaceId;
+        this.inProgress = [];
+        this.notSent = [];
+        this.activeChanges = [];
+        this.writeInProgress = false;
+        this.counter = 0;
+    };
+
+    /**
+     * Create and entry from an event and add it to active changes.
+     * An entry is of the form {"event": Blockly.Event, "entryId": string} where
+     * event is an event created on the workspace and entryId is of the form
+     * {workspaceId}:{counter}.
+     * @param {boolean} success Indicates the success of the database write.
```

Comment:
The param here is not the same name, nor the right type?

Reply:
Correct. Thanks for noting that.